pkg/controller/tas/node_failure_controller_test.go

```diff
@@ -0,0 +1,325 @@
+		t.Fatalf("Failed to get node %q: %v", nodeName, err)
+	}
+
+	updated := false
+	for i := range node.Status.Conditions {
+		if node.Status.Conditions[i].Type == corev1.NodeReady {
+			if node.Status.Conditions[i].Status != status {
+				node.Status.Conditions[i].Status = status
+				node.Status.Conditions[i].LastTransitionTime = metav1.NewTime(clock.Now())
+				updated = true
+			}
+			break
+		}
+	}
+	if !updated {
+		t.Logf("Node %q already had Ready condition status %q", nodeName, status)
+		return
+	}
+
+	if err := k8sClient.Status().Update(ctx, node); err != nil {
+		t.Fatalf("Failed to update node %q status: %v", nodeName, err)
+	}
+	t.Logf("Successfully updated node %q Ready condition to %s", nodeName, status)
+}
+
+func TestNodeFailureReconciler(t *testing.T) {
+	testStartTime := time.Now()
+	nodeNameUnhealthy := "test-node-unhealthy"
+	wlName := "test-workload"
+	nsName := "default"
+	wlKey := types.NamespacedName{Name: wlName, Namespace: nsName}
+
+	baseWorkload := utiltesting.MakeWorkload(wlName, nsName).
+		Finalizers(kueue.ResourceInUseFinalizerName).
+		ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+		Admitted(true).
+		Obj()
+
+	basePod := testingpod.MakePod("test-pod", nsName).
+		Annotation(kueuealpha.WorkloadAnnotation, wlName).
+		Label(kueuealpha.TASLabel, "true").
+		NodeName(nodeNameUnhealthy).
+		Obj()
+
+	tests := map[string]struct {
+		initObjs        []client.Object
+		req             reconcile.Request
+		advanceClock    time.Duration // Time to advance clock before reconcile
+		wantFailedNodes []string
+		wantRequeue     time.Duration
+	}{
+		"Node Found and Unhealthy (NotReady) - delay not passed": {
+			initObjs: []client.Object{
+				newNodeTest(nodeNameUnhealthy, corev1.ConditionFalse),
+				baseWorkload.DeepCopy(),
+				basePod.DeepCopy(),
+			},
+			req:             reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeNameUnhealthy}},
+			wantFailedNodes: nil,
+			wantRequeue:     NodeFailureDelay,
+		},
+		"Node Found and Unhealthy (NotReady) - delay passed": {
+			initObjs: []client.Object{
+				newNodeTest(nodeNameUnhealthy, corev1.ConditionFalse),
+				baseWorkload.DeepCopy(),
+				basePod.DeepCopy(),
+			},
+			req:             reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeNameUnhealthy}},
+			advanceClock:    NodeFailureDelay + time.Second,
+			wantFailedNodes: []string{nodeNameUnhealthy},
+		},
+		"Node Found and Healthy": {
+			initObjs: []client.Object{
+				newNodeTest(nodeNameUnhealthy, corev1.ConditionTrue),
+				utiltesting.MakeWorkload(wlName, nsName).
+					Finalizers(kueue.ResourceInUseFinalizerName).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+					Admitted(true).
+					Obj(),
+				basePod.DeepCopy(),
+			},
+			req:             reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeNameUnhealthy}},
+			wantFailedNodes: nil,
+		},
+	}
+
+	for name, tt := range tests {
```

Comment:
```suggestion
	for name, tc := range tests {
```
nit, just for consistency with other tests

Reply:
Done.